judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java

```diff
@@ -368,17 +378,17 @@ void testDeepCopyConstructor() {
         //When we add a composition Entity we must copy it, because that comp entity belong the created entity
 
         //Build entities for the test
-        entityD1 = entityDDao.create(EntityD.builder()
+        entityD1 = entityDDao.create(EntityDForCreate.builder()
                 .build());
-        entityD2 = entityDDao.create(EntityD.builder()
+        entityD2 = entityDDao.create(EntityDForCreate.builder()
                 .build());
-        singleRequiredConA = entityCDao.create(EntityC.builder()
+        singleRequiredConA = entityCDao.create(EntityCForCreate.builder()
                 .withStringC("C")
-                .withMultipleDonB(List.of(entityD1, entityD2))
+                .withMultipleDonB(List.of(entityD1.adaptTo(EntityDForCreate.class), entityD2.adaptTo(EntityDForCreate.class)))
```

Comment:
What if the old version remained alongside the new one? Thus, we don't have to use the adaptTo method.

Reply:
I think it's better to use only the 'EntityDForCreate' instance because composition always creates a new element. You can use 'adaptTo' if the composition doesn't contain the required association relation or 'EntityDForCreate.builderFrom(entityD1)./required relations/.build().'